src/structures/GuildChannel.js

```diff
@@ -357,7 +357,7 @@ class GuildChannel extends Channel {
     const newData = await this.client.api.channels(this.id).patch({
       data: {
         name: (data.name || this.name).trim(),
-        type: data.type ? ChannelTypes[data.type.toUpperCase()] : this.type,
+        type: ChannelTypes[data.type?.toUpperCase()],
```

Comment:
Wouldn't this trigger `ChannelTypes[undefined]` when type is not specified? If you wanted to pass a falsy value (including `0`), you could do:

```suggestion
        type: data.type ? ChannelTypes[data.type.toUpperCase()] : data.type,
```

Defaulting to whichever `data.type`'s value is when it's not truthy.

Reply:
It doesn't matter that it'd call `ChannelTypes[undefined]`, since that, in turn, returns `undefined`, thus not being sent to Discord. You could argue we don't support numbers as types, but that should come if we ever replace string types with numbers